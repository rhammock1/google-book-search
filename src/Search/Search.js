import React from 'react';

const Search = function () {
  return (
    <form className='search-form'>
      <label htmlFor='search'>Search:</label>
      <input id='search' name='search' type='text' placeholder='e.g. The Island' required />
      <button type='click'>Search</button>
    </form>
    
    )
}

export default Search;