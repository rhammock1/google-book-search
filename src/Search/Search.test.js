import React, { createElement } from 'react';
import ReactDOM from 'react-dom';
import Search from './Search';
import renderer from 'react-test-renderer';


describe('Search component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });
})