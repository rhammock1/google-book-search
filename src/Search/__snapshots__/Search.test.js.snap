// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders the UI as expected 1`] = `
<div
  className="search-bar"
>
  <label
    htmlFor="search"
  >
    Search:
  </label>
  <input
    id="search"
    name="search"
    placeholder="e.g. The Island"
    required={true}
    type="text"
  />
  <button
    type="click"
  >
    Search
  </button>
</div>
`;
